fastify.js

```diff
@@ -431,6 +436,9 @@ function build (options) {
   }
 
   function onBadUrl (path, req, res) {
+    if (options.frameworkErrors) {
+      return options.frameworkErrors(new FST_ERR_BAD_URL(path), req, res)
```

Comment:
as suggested by delvedor, here we could manage the req and res object like this the following code.

In this way, we will provide a solid and plain API to all the developers, where the `frameworkErrors` function will receive every time the same input: one fastify error, the enhanced request and the reply.

https://github.com/fastify/fastify/blob/911c8c015870555e5df2d78e3cc49687722e429e/lib/fourOhFour.js#L167-L182

The main goal would be the logger: everyone wants to log the errors, and if we deliver the plain HTTP-REQ and HTTP-RES the user will have issue to archive this

You should have all the parameters you need in that scope to reflect that logic 💪

Reply:
I believe I have everything i need in the scope.
So basically after those lines I should pass the newly created request and reply to the frameworkErrors function to expose them to the user right?

Reply:
can you also do:

```js
const frameworkErrors = options.frameworkErrors

// in the function

if (frameworkErrors) {
  return frameworkErrors(new FST_ERR_BAD_URL(path), req, res)
}
```

This will avoid `frameworkErrors` to be called with `this` as the `options` object.